server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/ContinuousComputation.java

```diff
@@ -49,6 +49,14 @@ public void onNewInput(T input) {
         }
     }
 
+    /**
+     * enqueues input if {@code expectedLatestKnownInput} is the latest known input
+     */
+    protected boolean compareAndEnqueue(T expectedLatestKnownInput, T input) {
+        assert expectedLatestKnownInput != null;
+        return enqueuedInput.compareAndSet(expectedLatestKnownInput, input);
```

Comment:
I think we need to schedule the `processor` instance? [Like onNewInput above](https://github.com/elastic/elasticsearch/blob/91474289122fc610b457b2836f6daafb60e44bd7/server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/ContinuousComputation.java#L48). Otherwise it won't run on the new input, I think 🤨

Reply:
I don't think so. The line you're referring to makes sure we start running a computation if there is none in progress since that is how the next one usually starts (if it is enqueued before the current one finishes). The new method should be used only if the current input `expectedLatestKnownInput` is not null which we have an assert for. I think similar to `onNewInput`, what is missing is a check for when asserts are disabled, i.e. a `Objects.requireNonNull`. I'll add that, and an extra comment.

Reply:
I've pushed fbbe084c3e5811dfb935d4fbb38869734412d4e7.